app/src/main/java/org/covidwatch/android/ui/util/DateFormatter.kt

```diff
@@ -0,0 +1,19 @@
+package org.covidwatch.android.ui.util
+
+import java.text.SimpleDateFormat
+import java.util.*
+
+object DateFormatter {
+    private const val DATE_PATTERN = "MMM dd, yyyy"
+
+    private var locale = Locale.getDefault()
+    private var date = SimpleDateFormat(DATE_PATTERN, locale)
+        get() {
+            if (locale == Locale.getDefault()) {
+                return field
+            }
+            return SimpleDateFormat(DATE_PATTERN, locale).also { field = it }
+        }
```

Comment:
can we reduce this to
```
private val dateFormat: SimpleDateFormat by lazy {
  SimpleDateFormat(DATE_PATTERN, Locale.getDefault())
}
```
?

Reply:
No because it's about the lazy initialization but about checking if the locale has changed every time this formatter is used. 
And. actually, I need to update the `locale` field when we create new `SimpleDateFormat` instance. Will change that in a sec :rocket: